parser/SqlParser.ypp

```diff
@@ -1648,6 +1649,16 @@ literal_value:
       YYERROR;
     }
   }
+  | TOKEN_INTERVAL TOKEN_STRING_SINGLE_QUOTED TOKEN_YEAR {
```

Comment:
Let's support all the datetime keywords. One way to do it:
```
| TOKEN_INTERVAL TOKEN_STRING_SINGLE_QUOTED datetime_unit {
  ...
}

datetime_unit:
  TOKEN_YEAR {
    $$ = new quickstep::ParseString(@1.first_line, @1.first_column, std::string("year"));
  }
  | TOKEN_MONTH {
    $$ = new quickstep::ParseString(@1.first_line, @1.first_column, std::string("month"));
  }
  ...
```
where `datetime_unit` is in the type group `%type <string_value_>`